lib/user.rb

```diff
@@ -0,0 +1,19 @@
+class User
+  attr_reader :name, :jokes, :jokes_id
+
+  def initialize(name) 
+    @name = name
+    @jokes = []
+    @jokes_id = []
+  end
+
+  def learn(joke)
+    @jokes_id << joke.id
+    @jokes << joke 
+  end
+
+  def tell(friend, joke)
+    friend.jokes << joke 
```

Comment:
since this is essentially your friend learning a joke, and your friend should be an instance of the user class (the interaction pattern tells us this), you can just call `friend.learn(joke)` here instead of doing the work over again.